2-Draft-Accepted/RFCNNNN-Native-Markdown-Rendering.md

```diff
@@ -0,0 +1,199 @@
+| Setext Header 2 | ![](../assets/MarkdownRendering/SetextHeader2-MD.png) | ![](../assets/MarkdownRendering/SetextHeader2.png) | ESC[4;93mHeader 1ESC[0m | ESC[4;33mHeader 1ESC[0m |
+
+#### Code Blocks
+
+Code blocks will be rendered with a lighter background color and a dark foreground text.
+Background color will be applied for the console width.
+No syntax highlighting will be done.
+`ESC[500@` is used to have sufficiently wide background color.
+
+| Markdown | Rendered | Escape Sequences |
+|-------------|----------|----------|
+| ![](../assets/MarkdownRendering/CodeBlock-MD.png) | ![](../assets/MarkdownRendering/CodeBlock.png) | ESC[48;2;155;155;155;38;2;30;30;30mTextESC[500@ESC[0m |
+
+### Span Elements
+
+Span elements will have varying degree of support depending on element type.
+
+#### Links
+
+Links in paragraphs will add double quotes around the link label and append the link URL surrounded by parenthesis.
+The link will be colored to differentiate from plain text.
+
+| Markdown | Rendered | Escape Sequences |
+|----------|----------|-------------|
+| ![](../assets/MarkdownRendering/Link-MD.png) | ![](../assets/MarkdownRendering/Link.png) | ESC[4;34m( )ESC[0m |
+
+If the links are part of a container like a pipe table, block quote or list; then just the link label will be shown.
+
+#### Emphasis
+
+Emphasis will be rendered using escape sequence `ESC[1mBold TextESC[0m` and `ESC[36mItalics TextESC[0m` for bold text and italics text respectively.
+
+| Element | Markdown | Rendered | Escape Sequences |
+|---------|-------------|----------|----------|
+| Bold | ![](../assets/MarkdownRendering/Bold-MD.png) | ![](../assets/MarkdownRendering/Bold.png) | ESC[1mBold TextESC[0m |
+| Italics | ![](../assets/MarkdownRendering/Italics-MD.png) | ![](../assets/MarkdownRendering/Italics.png)| ESC[36mItalics TextESC[0m |
+
+#### Code
+
+Inline code elements will be rendered similar to code blocks, with the exception on background color to not extend beyond the markdown element.
+
+|  Markdown | Rendered | Escape Sequences |
+|-------------|----------|----------|
+| ![](../assets/MarkdownRendering/Code-MD.png) | ![](../assets/MarkdownRendering/Code.png) | ESC[48;2;155;155;155;38;2;30;30;30mTextESC[0m`|
+
+#### Images
+
+The alt-text of the image will be surrounded by `[` and `]` and colorized to differentiate from plain text.
+Escape code for images alt-text
+
+| Markdown | Rendered | Escape Sequences |
+|-------------|----------|----------|
+| ![](../assets/MarkdownRendering/Image-MD.png) | ![](../assets/MarkdownRendering/Image.png) | ESC[33m[alt-text]ESC[0m |
+
+## Future Work
+
+`Paragraphs and Line Breaks`, `Block Quotes`, `Lists`, `Horizontal Rules`, `Pipe tables` and `HTML code` will be considered for future versions and rendered as plain text in this version.
+
+## Alternate Proposals and Considerations
+
+Introduce a new sigil to define a markdown string.
+This would be later rendered using formating files.
```

Comment:
Have we plan to add [MarkDown] type to automatically format output strings?
```powershell
[MarkDown] $var = " ... "
$var
$var | Out-Default
```

Reply:
Adding a type accelerator `[Markdown]` is not in scope for this RFC.

Reply:
My comment is not about the accelerator - it is about the type so we can use ETS and auto-formatting.

Reply:
Yes, the cmdlet returns a strongly type object. I will add a line about it.